### ISSUE TITLE
MVK test with F compsets using mpas sea ice

### ISSUE THREAD

**Issue by mt5555** [E3SM-Project/E3SM#4645]
MVK test uses the cime ensemble capability and runs 30 ensemble members.  If one configures a task with each ensemble using 24 cores, we have NTASKS=720, and NTASKS_PER_INST=24.   This causes MPAS-I to look for a decomposition file for 720 tasks instead of 24 tasks.   If the 720 file doesn't exist, ./create_test fails.   

Example:

./create_test MVK_P24x1.ne4_oQU240.F2010 -q short -o -g -b master --baseline-root ~/scratch1/baselines/intel

Error:
```
inished RUN for test MVK_P24x1.ne4_oQU240.F2010.compy_intel in 3.962719 seconds (FAIL). [COMPLETED 1 of 1]
    Case dir: /compyfs/tayl519/e3sm_scratch/MVK_P24x1.ne4_oQU240.F2010.compy_intel.G.20211108_112453_ru4oyl
    Errors were:
        env_batch.xml appears to have changed, regenerating batch scripts
        manual edits to these file will be lost!
        
        Model elm no file specified for finidat
        Model elm no file specified for finidat
        wget failed with output:  and errput --2021-11-08 11:36:25--  https://web.lcrc.anl.gov/public/e3sm/inputdata/ice/mpas-cice/oQU240/mpas-cice.graph.info.151209.part.720
        Resolving web.lcrc.anl.gov (web.lcrc.anl.gov)... 140.221.70.30
        Connecting to web.lcrc.anl.gov (web.lcrc.anl.gov)|140.221.70.30|:443... connected.
        HTTP request sent, awaiting response... 404 Not Found
        2021-11-08 11:36:25 ERROR 404: Not Found.
        
        Model elm no file specified for finidat
        svn export failed with output:  and errput svn: E170000: URL 'https://svn-ccsm-inputdata.cgd.ucar.edu/trunk/inputdata/ice/mpas-cice/oQU240/mpas-cice.graph.info.151209.part.720' doesn't exist
        
        ERROR: Could not find all inputdata on any server

```

relevant section from env_mach_pes.xml file: (from copy)

```
    <entry id="NTASKS">
      <type>integer</type>
      <values>
        <value compclass="ATM">720</value>
        <value compclass="CPL">720</value>
        <value compclass="OCN">720</value>
        <value compclass="WAV">720</value>
        <value compclass="GLC">720</value>
        <value compclass="ICE">720</value>
        <value compclass="ROF">720</value>
        <value compclass="LND">720</value>
        <value compclass="ESP">720</value>
        <value compclass="IAC">720</value>
      </values>
      <desc>number of tasks for each component</desc>
    </entry>
    <entry id="NTASKS_PER_INST">
      <type>integer</type>
      <values>
        <value compclass="ATM">24</value>
        <value compclass="OCN">24</value>
        <value compclass="WAV">24</value>
        <value compclass="GLC">24</value>
        <value compclass="ICE">24</value>
        <value compclass="ROF">24</value>
        <value compclass="LND">24</value>
        <value compclass="ESP">24</value>
        <value compclass="IAC">24</value>
      </values>
      <desc>Number of tasks per instance for each component. DO NOT EDIT: Set automatically by case.setup b$
   ```

**Comment by rljacob** [comment]
@wlin7 did we decide that these tests would be changed to use CICE ?

**Comment by mt5555** [comment]
Using F2010-CICE does avoid the problem (and does appear to be running on Compy).  

However, i believe this indicates a bug in CIME's ensemble capability interacting with MPAS.  I think MPAS is unique in that it has input files that depend on number of tasks, and that calculation needs to use  NTASKS_PER_INST instead of NTASKS

**Comment by wlin7** [comment]
F2010 is also ok now. 

The required oQU240/mpas-cice.graph.info.151209.part.720 is available on lcrc's inputdata path, but was not on the inputdata server.

**Comment by mt5555** [comment]
The error looking for part.720 is a red herring - it's not needed and cime shouldn't be trying to download it (and we dont need to add it to the inputdata server)

The case configuration script needs to be looking for part.24 instead of part.720.  (720 = NIST*24)

I haven't tested this, but suspect this would impact all NIST>1 cases with mpas components.

**Comment by rljacob** [comment]
Yes MPAS components do not play well with multi-instance.  cc:  @jonbob .  So we will have to use the F2010-CICE compset.

**Comment by jonbob** [comment]
@rljacob - please let me know if this becomes critical and I'll dig into it

**Comment by mt5555** [comment]
not critical.

I think we just need a one line fix (below). testing now...  

```
diff --git a/components/mpas-seaice/cime_config/buildnml b/components/mpas-seaice/cime_config/buildnml
index 486683f..f2d30e5 100755
--- a/components/mpas-seaice/cime_config/buildnml
+++ b/components/mpas-seaice/cime_config/buildnml
@@ -40,7 +40,7 @@ def buildnml(case, caseroot, compname):
     #ninst_ice       = case.get_value("NINST_ICE")
     ninst_ice        = 1 # Change if you want multiple instances... though this isn't coded yet.
     ninst_ice_real   = case.get_value("NINST_ICE")
-    ntasks_ice       = case.get_value("NTASKS_ICE")
+    ntasks_ice       = case.get_value("NTASKS_PER_INST_ICE")
     rundir           = case.get_value("RUNDIR")
     run_type         = case.get_value("RUN_TYPE")
     run_refcase      = case.get_value("RUN_REFCASE")
```

**Comment by jonbob** [comment]
OK, please let me know

**Comment by mt5555** [comment]
That fix above (using NTAKS_PER_INST_ICE instead of NTASKS_ICE) works, and sets the correct partition file for sea ice when using the F2010 compset.  

I think it's correct, it wont  affect our normal usage (NINST=1), so I'll go ahead and submit a PR.

**Comment by mt5555** [comment]
fixed in #4654
